src/util/RtreeBuild.cpp

```diff
@@ -0,0 +1,1133 @@
+          minXB0 = std::midpoint(minElement.box.min_corner().get<0>(), minElement.box.max_corner().get<0>());
+          maxXB0 = std::midpoint(currentLastElement.box.min_corner().get<0>(), currentLastElement.box.max_corner().get<0>());
+
+          minXB1 = std::midpoint(currentElement.box.min_corner().get<0>(), currentElement.box.max_corner().get<0>());
+          maxXB1 = std::midpoint(maxElement.box.min_corner().get<0>(), maxElement.box.max_corner().get<0>());
+        } else {
+          minYB0 = std::midpoint(minElement.box.min_corner().get<1>(), minElement.box.max_corner().get<1>());
+          maxYB0 = std::midpoint(currentLastElement.box.min_corner().get<1>(), currentLastElement.box.max_corner().get<1>());
+
+          minYB1 = std::midpoint(currentElement.box.min_corner().get<1>(), currentElement.box.max_corner().get<1>());
+          maxYB1 = std::midpoint(maxElement.box.min_corner().get<1>(), maxElement.box.max_corner().get<1>());
+        }
+
+        currentlyAtSTimesI = false;
+      } else {
+        break;
+      }
+
+      Rtree::BoundingBox b0 = Rtree::createBoundingBox(minXB0, minYB0, maxXB0, maxYB0);
+      Rtree::BoundingBox b1 = Rtree::createBoundingBox(minXB1, minYB1, maxXB1, maxYB1);
+
+
+      double cost = costFunctionTGS(b0, b1, dim);
+
+      if (splitResult.bestCost == -1 || cost < splitResult.bestCost) {
+        splitResult.bestCost = cost;
+        splitResult.bestDim = dim;
+        splitResult.bestLastElement = currentLastElement;
+        splitResult.bestElement = currentElement;
+        splitResult.bestMinElement = minElement;
+        splitResult.bestMaxElement = maxElement;
+        splitResult.bestIndex = i;
+      }
+    }
+    currentlyAtSTimesI = false;
+  }
+
+  return splitResult;
+}
+
+std::pair<OrderedBoxes, OrderedBoxes> OrderedBoxes::SplitAtBest(const std::string& filePath, size_t S, size_t M, uint64_t maxBuildingRamUsage) {
+  if (this->workInRam) {
+    return this->SplitAtBestInRam(S, M);
+  } else {
+    return this->SplitAtBestOnDisk(filePath, S, M, maxBuildingRamUsage);
+  }
+}
+
+std::pair<OrderedBoxes, OrderedBoxes> OrderedBoxes::SplitAtBestInRam(size_t S, size_t M) {
+  /**
+     * Split the ordered boxes in ram. First determine the best split and then perform it
+   */
+
+  struct SplitResult splitResult = this->GetBestSplit();
+
+  OrderedBoxes split0;
+  OrderedBoxes split1;
+
+  struct SplitBuffersRam splitBuffers;
+
+  splitBuffers.s0Dim0 = std::make_shared<multiBoxWithOrderIndex>();
+  splitBuffers.s0Dim1 = std::make_shared<multiBoxWithOrderIndex>();
+  splitBuffers.s1Dim0 = std::make_shared<multiBoxWithOrderIndex>();
+  splitBuffers.s1Dim1 = std::make_shared<multiBoxWithOrderIndex>();
+
+  splitBuffers.s0SmallDim0 = std::make_shared<multiBoxWithOrderIndex>();
+  splitBuffers.s0SmallDim1 = std::make_shared<multiBoxWithOrderIndex>();
+  splitBuffers.s1SmallDim0 = std::make_shared<multiBoxWithOrderIndex>();
+  splitBuffers.s1SmallDim1 = std::make_shared<multiBoxWithOrderIndex>();
```

Comment:
Doesn't the constructor of `splitBuffers` already take care of this initialization, and thus these lines can go?
(If not, then it definitely should).